### PR TITLE
Purescript 0.14 compatibility, bridging of nested types, bridging of `Map`

### PR DESCRIPTION
**Pull request by peterbecich**
https://github.com/eskimor/purescript-bridge/issues/55

This PR makes the library compatible with Purescript 0.14.

If https://github.com/coot/purescript-argonaut-aeson-generic/pull/15 is merged, this should be updated: 
https://github.com/eskimor/purescript-bridge/blob/6d95574c299612b829c5b9a8925035ec4e9158f4/example/packages.dhall#L16-L19

This PR also provides a bridge between a Haskell `Map` and a Purescript `Object`. The example includes this:
https://github.com/eskimor/purescript-bridge/blob/6d95574c299612b829c5b9a8925035ec4e9158f4/example/src/Types.hs#L23
https://github.com/eskimor/purescript-bridge/blob/6d95574c299612b829c5b9a8925035ec4e9158f4/example/src/Types.purs#L23

### CODE REVIEW

**Review by eskimor — APPROVED** [review]

File: src/Language/PureScript/Bridge/Printer.hs [hunk 1]
```diff
@@ -69,45 +69,63 @@ moduleToText settings m = T.unlines $
...
 _lensImports :: Switches.Settings -> [ImportLine]
 _lensImports settings
   | Switches.generateLenses settings =
-    [ ImportLine "Data.Maybe" $ Set.fromList ["Maybe(..)"]
-    , ImportLine "Data.Lens" $ Set.fromList ["Iso'", "Prism'", "Lens'", "prism'", "lens"]
-    , ImportLine "Data.Lens.Record" $ Set.fromList ["prop"]
-    , ImportLine "Data.Lens.Iso.Newtype" $ Set.fromList ["_Newtype"]
-    , ImportLine "Data.Symbol" $ Set.fromList ["SProxy(SProxy)"]
-    , ImportLine "Data.Newtype" $ Set.fromList ["class Newtype"]
+    [ ImportLine "Data.Lens" Nothing $ Set.fromList ["Iso'", "Prism'", "Lens'", "prism'", "lens"]
```
Comment (peterbecich) [inline src/Language/PureScript/Bridge/Printer.hs:89]: alphabetized imports

**Comment by peterbecich** [comment]
@eskimor , please review again.

I believe this PR fixes or implements:
- Purescript 0.14 compatibility
- bridging of `Map` type
- bridging of nested types

I have tested it with the example in this repo, and here: https://github.com/peterbecich/halogen-chess/
In the latter example, Haskell types `Sq` and `Move` are correctly bridged to Purescript: https://github.com/peterbecich/halogen-chess/blob/582ff4f9acf32c2b89653e88d1df1ae43725a6cd/src/Game/Chess/Move.purs#L23-L28

**Comment by eskimor** [comment]
Thank you @peterbecich - I will have a closer look asap!

**Comment by peterbecich** [comment]
Similar to https://github.com/input-output-hk/purescript-bridge/pull/3 and https://github.com/input-output-hk/purescript-bridge/pull/4

**Pull request merged**